Document business connections (#1034)
grammyjs/website: site/docs/advanced/business.md

# Telegram Business

> Telegram Business allows your private chat with another (human) user to be managed by a bot

Reviewer: Punc

This includes sending and receiving messages on your behalf.

> Typically, this is useful if you run your business on Telegram, and that other user is a customer.

Reviewer: ```suggestion
Typically, this is useful if you run your business on Telegram, and that other users are customers.
```

Author: We talk about the chat between the reader and a different user. This sentence refers to that other user in said chat, not to any number of people.

> If you are not familiar with Telegram Business yet, check out the [official docs](https://core.telegram.org/bots#manage-your-business) by Telegram before you continue.

Naturally, grammY has full support for this.

## Handling Business Messages

> A bot can manage a private chat between two users via Telegram Business.

Reviewer: Optional suggestion. Resolve without commenting.

```suggestion
A bot can manage a private chat between two users via Telegram Business—an account that is subscribed to Telegram's business subscription.
```

# Telegram Business

> Telegram Business allows your private chat with another (human) user to be managed by a bot.

Author: Yes, later

> Telegram Business allows your private chat with another (human) user to be managed by a bot.

Author: done

Typically, this is useful if you run your business on Telegram, and that other user is a customer.

> If you are not familiar with Telegram Business yet, check out the [official docs](https://core.telegram.org/bots#manage-your-business) by Telegram before you continue.

Naturally, grammY has full support for this.

## Handling Business Messages

A bot can manage a private chat between two users via Telegram Business.
This is done via a _business connection_ object that looks like [this](/ref/types/businessconnection).

### Receiving Business Messages

Once a business connection was set up, the bot will **receive messages** from _both chat participants_.

```ts
bot.on("business_message", async (ctx) => {
  // Access the message object.
  const message = ctx.businessMessage;
  // Shortcuts work as expected.

>   const m = ctx.msg;

Reviewer 2: maybe it is fine. but i would prefer making same as other docs examples.

>   const m = ctx.msg;

Reviewer 2: okay

```ts
bot.on("business_message", async (ctx) => {
  // Get information about the business connection.
  const conn = await ctx.getBusinessConnection();
  const employee = conn.user;
  // Check who sent this message.
  if (ctx.from.id === employee.id) {
    // You sent this message.
  } else {
    // Your customer sent this message.
  }
});
```

You can also skip calling `getBusinessConnection` for every update by doing [this](#working-with-business-connections).

### Sending Messages

Your bot is able to **send messages** to this chat _without being a member of the chat_.
It works as expected with `ctx.reply` and all of its variants.
grammY checks if the [context shortcut](../guide/context#shortcuts) `ctx.businessConnectionId` is available, so it can send the message to the managed business chat.

```ts
bot.on("business_message").filter(
  async (ctx) => {
    const conn = await ctx.getBusinessConnection();
    return ctx.from.id !== conn.user.id;
  },
  async (ctx) => {
    // Automatically respond to all customer questions.
    if (ctx.msg.text.endsWith("?")) {
      await ctx.reply("Soon.");
    }
  }
);
```

This will look as if you had sent the message yourself.

> There is no way to tell for you or your customer whether the message was sent manually or via your bot.

Author: Like this?
```suggestion
There is no way to tell for your customer whether the message was sent manually or via your bot.
(You will see a small indicator for this, though.)
```

(But your bot is probably much faster at replying than you.
Sorry.)

## Going Further

There are a few more things to consider when integrating your bot with Telegram Business.
We will cover a few aspects here briefly.

> ### Editing or Deleting Business Messages

Author: I believe that people will ask “how do I edit or a delete a message in the managed chat” so I want to talk explicitly about editing or deleting messages (and say that it's impossible).

> ### Editing or Deleting Business Messages

Reviewer 2: okay
